test/cmocka/src/audio/google_rtc_audio_processing/google_rtc_audio_processing_process.c

```diff
@@ -0,0 +1,255 @@
+				   parameters->period_size);
+	google_rtc_audio_processing_state->out = create_test_sink(
+		google_rtc_audio_processing_state->dev, 0, SOF_IPC_FRAME_S16_LE,
+		parameters->num_output_channels, parameters->period_size);
+	google_rtc_audio_processing_state->mic->stream.rate =
+		parameters->sample_rate_hz;
+	google_rtc_audio_processing_state->ref->stream.rate =
+		parameters->sample_rate_hz;
+	google_rtc_audio_processing_state->out->stream.rate =
+		parameters->sample_rate_hz;
+	google_rtc_audio_processing_state->mic->source->ipc_config.type =
+		SOF_COMP_DAI;
+	google_rtc_audio_processing_state->ref->source->ipc_config.type =
+		SOF_COMP_DEMUX;
+
+	rc = comp_trigger(google_rtc_audio_processing_state->dev,
+			  COMP_TRIGGER_PREPARE);
+	assert_return_code(rc, 0);
+	rc = comp_prepare(google_rtc_audio_processing_state->dev);
+	assert_return_code(rc, 0);
+	rc = comp_trigger(google_rtc_audio_processing_state->dev,
+			  COMP_TRIGGER_PRE_START);
+	assert_return_code(rc, 0);
+	rc = comp_trigger(google_rtc_audio_processing_state->dev,
+			  COMP_TRIGGER_START);
+	assert_return_code(rc, 0);
+
+	/* assigns verification function */
+	google_rtc_audio_processing_state->verify = verify_s16_to_s16;
+
+	/* assign test state */
+	*state = google_rtc_audio_processing_state;
+	return 0;
+}
+
+static int test_teardown(void **state)
+{
+	struct google_rtc_audio_processing_test_state
+		*google_rtc_audio_processing_state = *state;
+	struct comp_dev *dev = google_rtc_audio_processing_state->dev;
+	int rc;
+
+	rc = comp_trigger(dev, COMP_TRIGGER_STOP);
+	assert_return_code(rc, 0);
+	rc = comp_trigger(dev, COMP_TRIGGER_RESET);
+	assert_return_code(rc, 0);
+	rc = comp_reset(dev);
+	assert_return_code(rc, 0);
+
+	comp_free(dev);
+
+	free_test_sink(google_rtc_audio_processing_state->mic);
+	free_test_sink(google_rtc_audio_processing_state->ref);
+	free_test_source(google_rtc_audio_processing_state->out);
+	test_free(google_rtc_audio_processing_state);
+
+	return 0;
+}
+
+#if CONFIG_FORMAT_S16LE
+static void fill_source_s16(struct google_rtc_audio_processing_test_state
+				    *google_rtc_audio_processing_state)
+{
+	int16_t *mic = NULL;
+	int16_t *ref = NULL;
+	uint32_t i;
+	uint32_t length = audio_stream_get_free_frames(
+		&google_rtc_audio_processing_state->mic->stream);
+	uint32_t length_bytes = audio_stream_get_free_bytes(
+		&google_rtc_audio_processing_state->mic->stream);
+
+	for (i = 0; i < length; ++i) {
+		mic = audio_stream_write_frag_s16(
+			&google_rtc_audio_processing_state->mic->stream, i);
+		ref = audio_stream_write_frag_s16(
+			&google_rtc_audio_processing_state->ref->stream, i);
+		*mic = *ref = 0;
+	}
+	comp_update_buffer_produce(google_rtc_audio_processing_state->mic,
+				   length_bytes);
+	comp_update_buffer_produce(google_rtc_audio_processing_state->ref,
+				   length_bytes);
+}
+
+static void verify_s16_to_s16(struct comp_dev *dev, struct comp_buffer *mic,
+			      struct comp_buffer *ref, struct comp_buffer *out)
+{
```

Comment:
The magic is missing :)

Reply:
@lkoenig just want to confirm this is still not running any actual data validation